src/Akka.Persistence.Sql/Query/QueryThrottler.cs

```diff
@@ -40,31 +59,72 @@ internal sealed class ReturnQueryStart
     private ReturnQueryStart() { }
 }
 
+#region Test classes
+
+/// <summary>
+/// For testing purposes
+/// </summary>
+internal sealed class GetUsedPermits
+{
+    public static readonly GetUsedPermits Instance = new();
+    private GetUsedPermits() { }
+}
+
+/// <summary>
+/// For testing purposes
+/// </summary>
+internal sealed class GetPendingRequests
+{
+    public static readonly GetPendingRequests Instance = new();
+    private GetPendingRequests() { }
+}
+
+internal sealed class GetWatchCount
+{
+    public static readonly GetWatchCount Instance = new();
+    private GetWatchCount() { }
+}
+
+#endregion
+
 /// <summary>
 /// Token bucket throttler that grants queries permissions to run each iteration
 /// </summary>
 /// <remarks>
-/// Works identically to the RecoveryPermitter built into Akka.Persistence.
+/// Works almost identically to the RecoveryPermitter built into Akka.Persistence.
+/// 
+/// NOTE: Since this permitter works with Ask and we can't death watch an Ask temp
+///       actor, there is a possibility that a permit requester failed to return
+///       the request permit.
+/// 
+///       ALWAYS USE A TRY...FINALLY BLOCK WHEN USING ASK AND RETURN THE PERMIT IN
+///       THE FINALLY BLOCK
 /// </remarks>
 internal sealed class QueryThrottler : ReceiveActor
 {
-    private readonly LinkedList<IActorRef> _pending = new();
+    private readonly LinkedList<PendingRequest> _pending = new();
     private readonly ILoggingAdapter _log = Context.GetLogger();
+    private long _watchCount;
     private int _usedPermits;
     private int _maxPendingStats;
 
     public QueryThrottler(int maxPermits)
     {
         MaxPermits = maxPermits;
         
-        Receive<RequestQueryStart>(_ =>
+        Receive<RequestQueryStart>(request =>
         {
-            Context.Watch(Sender);
+            if(Sender is ActorRefWithCell)
+            {
+                _watchCount++;
+                Context.Watch(Sender);
+            }
```

Comment:
Actual fix, never watch any `IActorRef` with no `ActorCell`, these actors are not compatible with death watch.